test/versioned/fastify/new-state-tracking.tap.js

```diff
@@ -83,6 +83,34 @@ tap.test('fastify with new state tracking', (t) => {
 
     t.equal(transactions.length, 2)
   })
+
+  t.test('should wrap request/response hooks', async (t) => {
+    let ok = false
+
+    fastify.addHook('onRequest', (request, reply, done) => {
```

Comment:
personally I would put these kind of tests in the naming suites.  I'd abstract away the creation of the fastify server and add hooks for every single request lifecycle event.  Then assert the segment names which means it's prob best to name each plugin function instead of arrow functions

Reply:
I ended up placing them in their own test file because it seems the hook isn't affected by the differences between v2 and v3. LMK if this is sub-optimal!